File: Exercises/tutorial2-parallel-programming/python/src/printer.py
from pystreamapi import Stream
import time


def print_value(v):
    print(v)


def main():
    array = list(range(1000000))

    print("Serial execution")
    start = time.time()
    Stream.of(array)
    end = time.time()
    print("Serial: ", end - start)

    #Stream.of(array).for_each(print_value)

    print("Parallel execution")
    start = time.time()
    Stream.of(array).parallel()
    end = time.time()
    print("Parallel: ", end - start)
    #Stream.of(array).parallel().for_each(print_value)


if __name__ == "__main__":
    main()

File: Exercises/tutorial2-parallel-programming/java/src/Printer.java
import java.util.Arrays;

public class Printer {
    public static void main(String[] args) {
        int[] array = new int[10];
        for (int i = 0; i < array.length; i++) {
            array[i] = i;
        }

        System.out.println("Serial execution");
        Arrays.stream(array)
                .forEach(v -> System.out.println(v));

        System.out.println("Parallel execution");
        Arrays.stream(array).parallel()
                .forEach(v -> System.out.println(v));
    }
}
